packages/sdk/src/utils/EventSubscription.ts

```diff
@@ -0,0 +1,206 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { DocumentAction } from '../types/ActionTypes';
+import { AnimationPlaybackType, FrameAnimationType } from '../types/AnimationTypes';
+import { BarcodeFrameValidationResult } from '../types/BarcodeTypes';
+import { CharacterStyle } from '../types/CharacterStyleTypes';
+import { DocumentColor } from '../types/ColorStyleTypes';
+import { ConfigType, RuntimeConfigType, SelectedLayoutFrame, Id, AsyncError } from '../types/CommonTypes';
+import { AuthCredentials, AuthRefreshRequest, ConnectorEvent, ConnectorInstance } from '../types/ConnectorTypes';
+import { UndoState } from '../types/DocumentTypes';
+import { DocumentFontFamily } from '../types/FontTypes';
+import { FrameLayoutType, Frame } from '../types/FrameTypes';
+import { LayoutPropertiesType, MeasurementUnit, LayoutListItemType } from '../types/LayoutTypes';
+import { PageSize } from '../types/PageTypes';
+import { ParagraphStyle } from '../types/ParagraphStyleTypes';
+import { CornerRadiusUpdateModel } from '../types/ShapeTypes';
+import { SelectedTextStyle } from '../types/TextStyleTypes';
+import { Variable } from '../types/VariableTypes';
+import { ViewMode } from '../types/ViewModeTypes';
+import { Viewport } from '../types/ViewportTypes';
+import { ToolType } from './Enums';
+
+class EventSubscriptionBase {}
+
+export class SingleSubscription<T extends (...args: any[]) => any> extends EventSubscriptionBase {
+    protected callbacks: Map<string, T> = new Map();
+    protected legacyEventHandler: (() => T | undefined) | undefined;
+
+    constructor(legacyEventHandler?: () => T | undefined) {
+        super();
+        this.legacyEventHandler = legacyEventHandler;
+    }
+
+    /**
+     * Executes the registered callbacks with the provided arguments.
+     *
+     * @param args The arguments to pass to the callbacks.
+     * @returns The result of executing the callbacks.
+     */
+    trigger(...args: Parameters<T>): any {
+        let result;
+
+        if (this.legacyEventHandler) {
+            const callback = this.legacyEventHandler();
+            if (callback) {
+                result = callback(...args);
+            }
+        }
+
+        for (const callback of this.callbacks.values()) {
```

Comment:
All of the event handlers are required to be sync? 
While the request handlers can be both async/sync?